src/system_data.jl

```diff
@@ -369,8 +383,10 @@ Removes all time series of a particular type from a System.
 """
 function remove_time_series!(data::SystemData, ::Type{T}) where {T <: TimeSeriesData}
     for component in iterate_components_with_time_series(data)
-        for ts in get_time_series_multiple(component, type = T)
-            remove_time_series!(data, typeof(ts), component, get_name(ts))
+        for ts_metadata in list_time_series_metadata(component)
```

Comment:
While debugging the issue I noticed that we were unnecessarily reading time series data from the file in order to remove them. I addressed that here.